### ISSUE TITLE
Cant get Real-Client-IP

### ISSUE THREAD

**Issue by koljakutschera** [containous/traefik#2311]
### Do you want to request a *feature* or report a *bug*?

Bug

### What did you do?

Using Docker image: traefik:alpine as main reverseproxy with following config:

```toml
checkNewVersion = true
logLevel = "ERROR"
defaultEntryPoints = ["https","http"]

[web]
address = ":8080"

[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
  [entryPoints.https.tls]

[retry]

[docker]
endpoint = "unix:///var/run/docker.sock"
domain = "docker.localhost"
watch = true
exposedbydefault = false

[acme]
email = "not@my.email"
storage = "acme.json"
entryPoint = "https"
OnHostRule = true
acmeLogging = true
```

for some containers with e.g. the following config:

```yml
      - "traefik.enable=true"
      - "traefik.example-node.backend=example-node"
      - "traefik.example-node.frontend.passHostHeader=true"
      - "traefik.example-node.port=80"
      - "traefik.example-node.docker.network=reverseproxy_default"
      - "traefik.example-node.frontend.rule=Host:not.my.domain"
```

### What did you expect to see?

Everything is working perfectly BUT when i log headers in my app-container i got:

```
,"x-forwarded-for":"172.16.0.1","x-forwarded-host":"not.my.domain","x-forwarded-proto":"https","x-forwarded-server":"xaassaxasxs"
```

172.16.0.1 is just the gateway of the reverseproxy-network.
But i expect to see the REAL-ip of the client.
I think in real-world-applications you need the real ips to e.g. rate-limit by ip...
I like traefik but this is a no-go.
I tried many different docker-network settings and `frontend.passHostHeader`... but traefik just wont expose the real-ips of the clients or did i miss something?

### What did you see instead?

IP of docker-network. Need real-client-ip

### Output of `traefik version`: (_What version of Traefik are you using?_)

1.4.0 alpine

Thank you very much.

**Comment by nmengin** [comment]
Hello @koljakutschera ,

Many thanks for your interest in the project.
But are you sure `172.16.0.1 is just the gateway of the reverseproxy-network`?

Because I reproduce an environment similar to yours in a compose file and the `X-forwarded-for` header I get in my backend contains the IP address from the client.

My configuration :

```yaml
version: '2'

services:
  proxy:
    image: traefik:1.4.0-alpine
    command: --web --docker --docker.domain=localhost.com --logLevel=DEBUG
    networks:
      - webgateway
    ports:
      - "80:80"
      - "8080:8080"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock

  whoami:
    image: emilevauge/whoami
    networks:
      - "webgateway"
    labels:
      - "traefik.backend=whoami"
      - "traefik.port=80"
      - "traefik.network=webgateway"
      - "traefik.frontend.rule=Host:my.localhost.com"

  curl-client:
    image: ubuntu
    networks:
      - "webgateway"
    command: sleep infinity

networks:
  webgateway:
    driver: bridge
```

When I execute the curl from a command line on my machine I've got this output :

```shell
[me@machine 756]$ curl http://my.localhost.com
Hostname: 95f5fae3c504
IP: 127.0.0.1
IP: 172.19.0.3
GET / HTTP/1.1
Host: my.localhost.com
User-Agent: curl/7.56.0
Accept: */*
Accept-Encoding: gzip
X-Forwarded-For: 172.19.0.1
X-Forwarded-Host: my.localhost.com
X-Forwarded-Port: 80
X-Forwarded-Proto: http
X-Forwarded-Server: 404e5b6445f4

[me@machine 756]$ ifconfig
br-0281eff9ff0d: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 172.19.0.1  netmask 255.255.0.0  broadcast 0.0.0.0
```

As you can see `172.19.0.1` is the IP address of my machine into Docker network and it's the value of `X-Forwarded-For` header too.

When I execute the curl command from the ubuntu service named `curl-client` in my configuration :

```shell
[me@machine 756]$ docker-compose exec curl-client apt-get update && docker-compose exec curl-client apt-get -y install curl && docker-compose exec curl-client curl http://172.19.0.2 -H host:my.localhost.com
Hostname: 52f294fdc093
IP: 127.0.0.1
IP: 172.19.0.4
GET / HTTP/1.1
Host: my.localhost.com
User-Agent: curl/7.47.0
Accept: */*
Accept-Encoding: gzip
X-Forwarded-For: 172.19.0.3
X-Forwarded-Host: my.localhost.com
X-Forwarded-Port: 80
X-Forwarded-Proto: http
X-Forwarded-Server: cf81fade021c

[me@machine 756]$ docker-compose exec curl-client cat /etc/hosts
127.0.0.1	localhost
::1	localhost ip6-localhost ip6-loopback
fe00::0	ip6-localnet
ff00::0	ip6-mcastprefix
ff02::1	ip6-allnodes
ff02::2	ip6-allrouters
172.19.0.3	402fb6ad12fb

```

As you can see `172.19.0.3` is the IP address of the service `curl-client` and it's the value of `X-Forwarded-For` header.

Moreover, if you update the image version to the last Træfik version `traefik:1.4.1-alpine`, the `X-real-IP` header will be passed to the backend too.

I hope I understood you use case and this information can help you.
I let you give us a feedback and close the issue if it's solved.

**Comment by koljakutschera** [comment]
Thank you @nmengin 

i was searching for the real-ip of the user which like you told is available now in v1.4.1. Just upgraded and x-real-ip header was added. Nice! 

Thx
